client/my-sites/sidebar/sidebar.jsx

```diff
@@ -611,8 +607,8 @@ export class MySitesSidebar extends Component {
 				</SidebarMenu>
 
 				<ExpandableSidebarMenu
-					onClick={ this.props.toggleMySitesSidebarSiteMenu }
-					expanded={ this.props.isSiteOpen }
+					onClick={ () => this.props.toggleSection( SIDEBAR_SECTION_SITE ) }
```

Comment:
Is defining an anonymous function within a prop value still considered a performance issue b/c of re-rendering? I noticed the recommendation to not do this has been removed from the [React.js docs on event handling](https://reactjs.org/docs/handling-events.html).

Reply:
Imho you're good - there's been lots of discussions around this and it highly depends on what you're doing. [Here's a link to what React docs say about that](https://reactjs.org/docs/faq-functions.html#is-it-ok-to-use-arrow-functions-in-render-methods). Also, here's an [in-depth article](https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578) by _@ryanflorence about this.

@sgomes and @jsnajdr may be able to provide more calypso-focused feedback.

Reply:
I'm not aware of anything specific about inline functions. As far as I know, the only potential problem with this sort of code is that you're creating a new instance of the function on every render, and thus causing a re-render of the subcomponent.

But that's not necessarily a problem in itself. Unless it's a super obvious performance issue (like writing an algorithm with exponential complexity), the general advice is to write your code the readable way, and only go back to improve performance if you measure an actual performance problem.

And if that's the case, feel free to ping me and I'd be happy to help you out 🙂 There are multiple ways of getting around this sort of issue (like memoization and `shouldComponentUpdate`) if it comes to that.

Reply:
In the specific case of this sidebar, inline event handlers don't matter much: the Sidebar component is connected to Redux, and therefore rerenders only when the props returned from `mapStateToProps` change. Therefore, if a rerender of `ExpandableSidebarMenu` is triggered, there's already a valid reason to do it (prop change) and the fact that there's one more changed prop (`onClick`) doesn't cause any extra work.

With all that, I think it's still not a bad habit to make the handlers constant:
```jsx
class Sidebar extends Component {
  toggleSiteSection = () => this.props.toggleSection( SIDEBAR_SECTION_SITE );
  toggleDesignSection = () => this.props.toggleSection( SIDEBAR_SECTION_DESIGN );

  render() {
    return (
      <Expandable onClick={ this.toggleSiteSection } />
      <Expandable onClick={ this.toggleDesignSection } />
    );
  }
}
```

Another, more elegant and less verbose approach is to use a function that returns a function and use `lodash.memoize` to remember the result:
```jsx
class Sidebar extends Component {
  toggleSection = memoize( id => () => this.props.toggleSection( id ) );

  render() {
    return (
      <Expandable onClick={ this.toggleSection( SIDEBAR_SECTION_SITE ) } />
      <Expandable onClick={ this.toggleSection( SIDEBAR_SECTION_DESIGN ) } />
    );
  }
}
```

But as I said, even the inline handlers are quite fine here.

Reply:
❤️ it, thanks for the knowledge share! I've updated this PR to use `memoize`, which is lightweight enough that to me it seems like a good habit for event handlers that need arguments.